tests/lib/ringbuffer/src/main.c

```diff
@@ -443,25 +476,76 @@ void test_reset(void)
 	zassert_true(granted == RINGBUFFER_SIZE - 1, NULL);
 }
 
+#ifdef CONFIG_64BIT
+static uint64_t ringbuf_stored[RINGBUFFER_SIZE];
+#else
+static uint32_t ringbuf_stored[RINGBUFFER_SIZE];
+#endif
+
+/**
+ * @brief verify the array stored by ringbuf
+ *
+ * @ingroup lib_ringbuffer_tests
+ *
+ * @see ring_buf_item_put(), ring_buf_item_get()
+ */
+void test_ringbuffer_array_perf(void)
+{
+	struct ring_buf buf_ii;
+	uint32_t input[3] = {0xaa, 0xbb, 0xcc};
+	uint32_t output[6] = {0xff, 0xff, 0, 0, 0, 0};
+	uint16_t type = 0;
+	uint8_t value = 0, size = 3;
+	int count = 0;
+	void *tp;
+
+	/*Init the ringbuf, and be stored in 64/32 bits array*/
+	ring_buf_init(&buf_ii, RINGBUFFER_SIZE, ringbuf_stored);
+
+	zassert_true(ring_buf_item_put(&buf_ii, 1, 2, input, 3) == 0, NULL);
+
+	/*verify the address stored by ringbuf is contiguous*/
+	for (int i = 0; i < 3; i++) {
+		zassert_equal(input[i], buf_ii.buf.buf32[i+1], NULL);
+	}
+
+	/*Offset to the end of the in-use data in the array*/
+	while (output[count] != 0) {
+		count++;
+	}
+	zassert_true(ring_buf_item_get(&buf_ii, &type, &value,
+				&output[count], &size) == 0, NULL);
+
+	/*verify input = output*/
+	for (int i = 0; i < 3; i++) {
+		zassert_equal(input[i], output[count + i], NULL);
+	}
+
+	/*the size of array stored by ringbuf is equal to the size of pointer*/
+	zassert_equal(sizeof(tp), sizeof(ringbuf_stored[0]), NULL);
+}
+
+
 /*test case main entry*/
 void test_main(void)
 {
 	ztest_test_suite(test_ringbuffer_api,
-			 ztest_unit_test(test_ringbuffer_init),/*keep init first!*/
-			 ztest_unit_test(test_ringbuffer_declare_pow2),
-			 ztest_unit_test(test_ringbuffer_declare_size),
-			 ztest_unit_test(test_ringbuffer_put_get_thread),
-			 ztest_unit_test(test_ringbuffer_put_get_isr),
-			 ztest_unit_test(test_ringbuffer_put_get_thread_isr),
-			 ztest_unit_test(test_ringbuffer_pow2_put_get_thread_isr),
-			 ztest_unit_test(test_ringbuffer_size_put_get_thread_isr),
-			 ztest_unit_test(test_ring_buffer_main),
-			 ztest_unit_test(test_ringbuffer_raw),
-			 ztest_unit_test(test_ringbuffer_alloc_put),
-			 ztest_unit_test(test_byte_put_free),
-			 ztest_unit_test(test_byte_put_free),
-			 ztest_unit_test(test_capacity),
-			 ztest_unit_test(test_reset)
-			 );
+		 ztest_unit_test(test_ringbuffer_init),/*keep init first!*/
+		 ztest_unit_test(test_ringbuffer_declare_pow2),
+		 ztest_unit_test(test_ringbuffer_declare_size),
```

Comment:
Please make your new changes conform to the whitespace conventions of the original so we can tell what changed.  If you want to reflow a big region like this, make that change a separate patch.

Reply:
Hi @andyross , thanks for your comments, you are right, I shouldn't change this, so I have regained this region.

But there are something confused me, I used my native tools to check this region and it showed me "line over 80 characters", so I modified it. Actually,I find that it can be pushed to github successfully without modifying this.